lib/galaxy/tools/__init__.py

```diff
@@ -2541,7 +2541,7 @@ def iterate(group_inputs, inputs, state_inputs, other_values=None):
             other_values = ExpressionContext( state_inputs, other_values )
             for input_index, input in enumerate( inputs.itervalues() ):
                 # create model dictionary
-                tool_dict = input.to_dict(trans)
+                tool_dict = input.to_dict(trans, other_values=other_values)
```

Comment:
@guerler This behavior has nothing to do with the datatype, but is due to the way that my particular tool's dynamically generated select lists work together.  I have not found another tool that includes 2 select lists like this.  The 3-tier dependency with tabular file at  https://gist.github.com/guerler/157b0deabadbb6a2f9f5 is different in that it does not contain a select list that is dynamically rendered based on the current selection of an associated dynamically rendered select list.

The underlying Galaxy tool form rendering framework should support the requirements of my tool.  From what I've discovered, the Galaxy framework does not support associated dynamically rendered select lists like this because the tool input parameters are not available when the tool model is being built.  Specifically, the iterate() method on about line #2540 in ~//lib/galaxy/tools/__init__.py does not receive the set of input parameters for the tool.  The other_values parameter is always None or an empty dictionary when passed to this method.

        # builds tool model including all attributes
        def iterate(group_inputs, inputs, state_inputs, other_values=None):

I believe this is a bug.  It seems that at some point when all of the many to_dict() methods for tool form components were written, a decision was made to not pass the tool's input parameters up through the many links in the chain of to_dict() methods, possibly because doing so posed problems when dealing with Grouping parameter types (e.g., repeat).  I was not involved in this decision or the to_dict() implementation, so I cannot be sure why, but this implementation decision ultimately results in the various get_html() methods in the ~/lib/galaxy/tools/parameters/basic.py code to not have access to tool input parameters that may be necessary for dynamic rendering.

I believe the fix for this bug is to discover why the Grouping parameter types break if the tool input parameters are available to the to_dict() methods when building the tool model and correct that issue.  Another possibility may be to filter out Grouping parameters from the other_values dictionary and pass all other input parameter types up through the chain of to_dict() methods.  But that seems like a band-aid approach to me.  However, I do not have all of the context regarding the how's and why's of this to_dict() implementation, so others may be able to provide better insight.